### ISSUE TITLE
OAuth2 client credentials flow not working for clients with multiple JWKs

### ISSUE THREAD

**Issue by oanam** [okta/okta-sdk-java#504]
The OAuth2 client credentials flow fails for clients with multiple JWK keys, when attempting to get an access token:
```
invalid_client - The client_assertion signature is invalid.
com.okta.sdk.impl.oauth2.OAuth2HttpException: invalid_client - The client_assertion signature is invalid.
	at com.okta.sdk.impl.oauth2.AccessTokenRetrieverServiceImpl.getOAuth2AccessToken(AccessTokenRetrieverServiceImpl.java:118)
	at com.okta.sdk.impl.oauth2.OAuth2ClientCredentials.eagerFetchOAuth2AccessToken(OAuth2ClientCredentials.java:44)
	at com.okta.sdk.impl.oauth2.OAuth2ClientCredentials.<init>(OAuth2ClientCredentials.java:37)
	at com.okta.sdk.impl.client.DefaultClientBuilder.build(DefaultClientBuilder.java:342)
```

It appears that the API call to /oauth2/v1/token works with no issues for clients that only have one JWK (regardless of whether the JWK has a kid or not).
However, based on my tests, when multiple JWKs are configured for the same client, the API appears to accept only client_assertion JWTs that include the kid of the signing key in the header.

Unfortunately, the SDK does not offer a way to add a header to the signed JWT, so attempts to get a token fail with the error above.

Here is the current implementation of the JWT signing in the SDK :
https://github.com/okta/okta-sdk-java/blob/master/impl/src/main/java/com/okta/sdk/impl/oauth2/AccessTokenRetrieverServiceImpl.java#L137
 
Based on my tests, this additional line (e.g. inserted at line 144), would solve the issue:
`.setHeaderParam("kid", kidOfPrivateKey) // missing from SDK`
 
I was able to sign tokens using the same code as AccessTokenRetrieverServiceImpl, plus the "kid" header, and the error no longer occurred.

Please implement a way of adding the kid of the signing key in the JWT header, if needed.

**Comment by sergiishamrai-okta** [comment]
Thank you for the request. This feature will be added. Please watch this issue for updates.

**Comment by arvindkrishnakumar-okta** [comment]
Fixed in v4.0.0
